src/main/java/uk/gov/register/serialization/RSFCreator.java

```diff
@@ -50,6 +51,30 @@ public RegisterSerialisationFormat create(Register register, int totalEntries1,
         return new RegisterSerialisationFormat(commands);
     }
 
+    public RegisterSerialisationFormat create(Register register, String indexName) {
+        Iterator<?> iterators = Iterators.concat(
+                register.getItemIterator(),
+                register.getDerivationEntryIterator(indexName));
+
+        Iterator<RegisterCommand> commands = Iterators.transform(iterators, obj -> (RegisterCommand) getMapper(obj.getClass()).apply(obj));
+        return new RegisterSerialisationFormat(commands);
+    }
+
+
+    public RegisterSerialisationFormat create(Register register, String indexName, int totalEntries1, int totalEntries2) {
+        Iterator<?> iterators;
+
+        if (totalEntries1 == totalEntries2) {
+            iterators = Iterators.singletonIterator(register.getRegisterProof(totalEntries1));
```

Comment:
Thats a great catch Michaela. What do you think of `iterators = Collections.<RegisterCommand>emptyList().iterator();` instead?